File: javascript/FizzBuzz/js/fizzbuzz.js
class FizzBuzzEngine {
    calculateNext(number) {
        var rules = [
            {divider: 15, result: "Fizz-Buzz"},
            {divider: 3, result: "Fizz"},
            {divider: 5, result: "Buzz"}
        ];

        var ruleFound = rules
            .find((rule) => FizzBuzzEngine.isDivisibleBy(number, rule.divider));

        return ruleFound != null
            ? ruleFound.result
            : number.toString();
    }

    static isDivisibleBy(number, divider) {
        return number % divider == 0;
    }
}

File: csharp/OutsideInDemo/Implementation/OutsideIn/FizzBuzz.cs
using System.Collections.Generic;
using System.Globalization;
using System.Linq;

namespace Implementation.OutsideIn
{
    public class FizzBuzz
    {
        private readonly IEnumerable<IRule> _rules;

        public FizzBuzz(IEnumerable<IRule> rules)
        {
            _rules = rules;
        }

        public FizzBuzz() : this(StandardFizzBuzzRules())
        {
        }

        private static IEnumerable<Rule> StandardFizzBuzzRules()
        {
            return new []
            {
                new Rule(15, "Fizz-Buzz"),
                new Rule(3, "Fizz"),
                new Rule(5, "Buzz"),
            };
        }

        public string CalculateFor(int number)
        {
            var result = _rules
                .Where(r => r.AppliesTo(number))
                .Select(r => r.Result)
                .FirstOrDefault();
            return result ?? number.ToString(CultureInfo.InvariantCulture);
        }
    }
}